robozonky-strategy-natural/src/main/antlr4/imports/MarketplaceFilters.g4

```diff
@@ -47,8 +47,8 @@ secondaryMarketplaceFilter returns [MarketplaceFilter result]:
 
 jointMarketplaceFilterConditions returns [Collection<MarketplaceFilterCondition> result]:
     { Collection<MarketplaceFilterCondition> result = new LinkedHashSet<>(); }
-    (c1=primaryMarketplaceFilterCondition { result.add($c1.result); } '; ')*
-    c2=primaryMarketplaceFilterCondition { result.add($c2.result); } DOT
```

Comment:
Was this a bug? (using primaryMarketplaceFilterCondition in jointMarketplaceFilterConditions)

Reply:
I think so. The primary conditions are a little bit more permissive, and since the joint filter was referencing primary conditions, it was allowing what it wasn't supposed to. Specifically filtering by remaining loan amount - which is impossible in secondary marketplace and in sell filters.